### PR TITLE
Add RunSpec onChange Trigger

### LINKED ISSUE #228: Add run spec change to Run Configuration change trigger
In addition to triggering runs when a pipeline version changes (#217), trigger a run when `spec.run` changes.

- Store `status.dependencies.runSpec.version` with hash of `spec.run`
- Trigger a run based on `status.triggers.runSpec.version`

### PR DESCRIPTION
**Pull request by jmendesky**
Closes #228

## Tasks

- [x] Implementation
- [x] Documentation updated

### CODE REVIEW

File: apis/pipelines/v1alpha5/run_types.go [hunk 1]
```diff
@@ -95,12 +95,24 @@ func WriteRunTimeParameters(oh pipelines.ObjectHasher, rps []RuntimeParameter) {
...
 }
 
+func (rs RunSpec) WriteRunSpec(oh pipelines.ObjectHasher) {
+	oh.WriteStringField(rs.Pipeline.String())
+	oh.WriteStringField(rs.ExperimentName)
+	WriteRunTimeParameters(oh, rs.RuntimeParameters)
+	pipelines.WriteKVListField(oh, rs.Artifacts)
+}
+
+func (rs RunSpec) ComputeVersion() string {
```
Comment (paoloambrosio-skyuk) [inline apis/pipelines/v1alpha5/run_types.go:105]: Wonder if we should have both `ComputeHash` and `CompuerVersion`?

File: controllers/pipelines/runconfiguration_controller_decoupled_test.go [hunk 2]
```diff
@@ -369,6 +369,31 @@ var _ = Describe("RunConfiguration controller k8s integration", Serial, func() {
 		})
 	})
 
+	When("RunSpec changes", func() {
+		It("onChange trigger creates a run when run template has changed", func() {
```
Comment (paoloambrosio-skyuk) [inline controllers/pipelines/runconfiguration_controller_decoupled_test.go:373]: Wish we had unit tests for this part to check that it is not triggering when it shouldn't.

File: apis/pipelines/v1alpha5/run_types.go (hunk 1 above)
Comment (paoloambrosio-skyuk) [reply apis/pipelines/v1alpha5/run_types.go:105]: Fine without it.

**Comment by jmendesky** [comment]
ok to test

**Review by paoloambrosio-skyuk — APPROVED** [review]

**Pull request merged**